src/go/wsl-helper/pkg/host/hostfile.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright © 2023 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package host
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+)
+
+const (
+	GatewayIP           = "192.168.1.2"
+	GatewayDomain       = "gateway.rancher-desktop.internal"
+	BeginConfig         = "# BEGIN Rancher Desktop configuration."
+	EndConfig           = "# END Rancher Desktop configuration."
+	DefaultHostFilePath = "/etc/hosts"
+)
+
+// AppendHostsFile reads the content of a host file
+// and appends a slice of entries to the file. The entries are
+// provided in the following format, e.g.:
+//
+//	newEntries := []string{
+//	  "127.0.0.1,example.com",
+//	  "127.0.0.1,another-example.com",
+//	}
+func AppendHostsFile(entries []string, hostsFilePath string) error {
+	exist, err := configExist(hostsFilePath)
```

Comment:
If the config exists but is wrong (e.g. we changed the IP address for some reason), this will skip editing the hosts file. That seems incorrect. But correct enough that we can do this in a follow-up PR, I think (because we always end up with a static value here).

Reply:
Filed this as #5795.